File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for (let l = 0; l < nums.length; l++) {
        for (let r = l + 1; r < nums.length; r++) {
            if (nums[l] + nums[r] === target) {
                return [l, r];
            }
        }
    }
    return null;
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i]; // 9 - 2 = 7, 9 -7 = 2

            if (map.containsKey(diff)) {
                return new int[] { map.get(diff), i };
            }

            map.put(nums[i], i);
        }
        return null;
    }
}